### ISSUE TITLE
Incorrect typings for client.getMessages

### ISSUE THREAD

**Issue by Loskir** [gram-js/gramjs#460]
`getMessages` is said to return `TotalList<Api.Message>`

https://github.com/gram-js/gramjs/blob/af4f9c2648bd4019b8824516cdb0482d011c5d61/gramjs/client/messages.ts#L661

However, it actually returns `TotalList<Api.Message | undefined>`. `undefined` is returned if the corresponding message id does not exist.

```ts
const r = await client.getMessages('@', {ids: [585, 586, 588]})
```

```
TotalList(3) [
  {
    // message
  },
  undefined,
  undefined,
  total: 3
]
```

**Comment by painor** [comment]
yes that's correct. but that only happens if you pass the "ids" param.

I remember trying to find a way to make the return type depends on wether ids param is present or not but I couldn't find any useful info.

**Comment by Loskir** [comment]
You can use function overloading to achieve it. I may be able to look into it later

**Comment by painor** [comment]
That would be cool. feel free to PR something or have an example code here so I can test it out.
